server/core/src/indexer/indexer.ts

```diff
@@ -607,23 +607,29 @@ export class FullTextIndexPipeline implements FullTextPipeline {
           if (this.cancelling) {
             return
           }
-          const newDocs: DocIndexState[] = (
-            await dbStorage.findAll<Doc>(
-              this.metrics,
-              c,
-              { _class: c, _id: { $nin: states } },
-              { limit: 1000, projection: { _id: 1, attachedTo: 1, attachedToClass: 1 } as any }
-            )
-          ).map((it) => {
-            return createStateDoc(it._id, c, {
-              stages: {},
-              attributes: {},
-              removed: false,
-              space: it.space,
-              attachedTo: (it as AttachedDoc)?.attachedTo ?? undefined,
-              attachedToClass: (it as AttachedDoc)?.attachedToClass ?? undefined
+          let newDocs: DocIndexState[] = []
+          try {
+            newDocs = (
+              await dbStorage.findAll<Doc>(
+                this.metrics,
+                c,
+                { _class: c, _id: { $nin: states } },
+                { limit: 500, projection: { _id: 1, attachedTo: 1, attachedToClass: 1 } as any }
+              )
+            ).map((it) => {
+              return createStateDoc(it._id, c, {
+                stages: {},
+                attributes: {},
+                removed: false,
+                space: it.space,
+                attachedTo: (it as AttachedDoc)?.attachedTo ?? undefined,
+                attachedToClass: (it as AttachedDoc)?.attachedToClass ?? undefined
+              })
             })
-          })
+          } catch (e) {
+            console.error(e)
```

Comment:
When it could fail?

Reply:
Mongo sometimes throw an error due a limit of 16 MB, I've reduced a limit in query a bit, and wrapped it in try-catch just in case

Reply:
This is strange, since 1000*3 fields should be less in any case, because of projection. ok.

Reply:
I'll check it after moving to migration, maybe there is some other bug that might show up somewhere else in future. Cause at least I have reproducible steps here